### PR TITLE
#5 & #6 Merchant items index

### LINKED ISSUE #50: Merchant Items index/show
6. Merchant Items Index Page

As a merchant,
When I visit my merchant items index page (merchants/:merchant_id/items)
I see a list of the names of all of my items
And I do not see items for any other merchant

7. Merchant Items Show Page

As a merchant,
When I click on the name of an item from the merchant items index page, (merchants/:merchant_id/items)
Then I am taken to that merchant's item's show page (/merchants/:merchant_id/items/:item_id)
And I see all of the item's attributes including:

- Name
- Description
- Current Selling Price

### PR DESCRIPTION
**Pull request by z-fitch**
__x__ Wrote Tests __x__ Implemented __x__ Reviewed

index page
Screenshot 2023-07-28 at 8 41 08 PM

show page
Screenshot 2023-07-28 at 9 07 37 PM

Necessary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally

Type of change

    [x] New feature
    [] Bug Fix

Implements/Fixes:

    Will add a merchant item index and show page closes #50

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [] No Tests have been changed
    [x] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [x] I have reviewed my code
    [x] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(REQUIRED)Please Include a link to a gif of your feelings about this branch

Link:
image

### CODE REVIEW

**Review by Daniel-Gallagher92 — APPROVED** [review]

File: app/models/item.rb [hunk 1]
```diff
@@ -2,4 +2,9 @@ class Item < ApplicationRecord
   belongs_to :merchant
   has_many :invoice_items
   has_many :invoices, through: :invoice_items
+
+  def current_price
+    price = (unit_price * 0.001).round(2)
```
Comment (ethanrossblack) [inline app/models/item.rb]: You're rounding by one too many 0's in your decimal. If the unit price is in cents, we should multiply by `0.01` to get the price in dollars, not by `0.00`=1`.

File: spec/features/merchants/merchant_items/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      expect(page).to have_link("#{@item_6.name}")
+      expect(page).to have_link("#{@item_7.name}")
+      expect(page).to have_link("#{@item_8.name}")
+      expect(page).to have_link("#{@item_9.name}")
+      expect(page).to have_link("#{@item_10.name}")
+    end
+  end
+
+  it "After clicking on a item name I am taken to the item show page" do
+    visit merchant_item_path(@merchant_1, @item_1.id)
```
Comment (ethanrossblack) [inline spec/features/merchants/merchant_items/show_spec.rb]: For this test, I think they want us to start on the merchant_items index page, then click a link to a merchant_item show page.

The validations all look good, though.

Maybe replace line 154 with 

```
visit merchant_items_path(@merchant_1)

click_link("#{@item_1.name}")

expect(current_path).to eq(merchant_item_path(@merchant_1, @item_1.id))
```

File: spec/models/item_spec.rb [hunk 3]
```diff
@@ -6,4 +6,17 @@
...
+
+
+  describe "instance methods" do 
+    describe "#currrent_price" do
+      it "returns the unit price as $xx.xx " do
+        merchant_3 = Merchant.create!(name: 'Willms and Sons')
+        item_1 = merchant_3.items.create!(name: 'Gold Ring', description: 'A ring thats gold', unit_price: 75107)
+        item_2 = merchant_3.items.create!(name: 'Silver ring', description: 'A ring thats silver', unit_price: 45107)
+        expect(item_1.current_price).to eq("$75.11")
+        expect(item_2.current_price).to eq("$45.11")
```
Comment (ethanrossblack) [inline spec/models/item_spec.rb]:
```suggestion
        expect(item_1.current_price).to eq("$751.07")
        expect(item_2.current_price).to eq("$451.07")
```

rounding error

File: spec/features/merchants/merchant_items/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+    within("div#merchant-item-header") do
+      expect(page).to have_content("#{@item_1.name}")
+    end
+
+    within("div#merchant-item-show") do
+      expect(page).to have_content("Description: #{@item_1.description}")
+      expect(page).to have_content("Current Price: $75.11")
```
Comment (ethanrossblack) [inline spec/features/merchants/merchant_items/show_spec.rb]:
```suggestion
      expect(page).to have_content("Current Price: $751.07")
```

rounding

**Review by ethanrossblack — CHANGES_REQUESTED** [review]
Looks very good! Just a few small things, the biggest being the function that turns the unit price into a clear dollar amount.

File: app/models/item.rb (hunk 1 above)
Comment (Daniel-Gallagher92) [reply app/models/item.rb]: If you look in the Mod 2 cheat sheet bookmarked in the slack channel, there is actually a method specifically for converting to dolla dolla bills y'all!!

Comment (z-fitch) [reply app/models/item.rb]: I fixed it, I tried the method in the cheat sheet and it gave me thousands :/

File: spec/features/merchants/merchant_items/show_spec.rb (hunk 2 above)
Comment (z-fitch) [reply spec/features/merchants/merchant_items/show_spec.rb]: Good point, I moved that test block and added ur suggestions

**Review by ethanrossblack — APPROVED** [review]
looks good! thanks for making those changes

**Pull request merged**